packages/story-editor/src/components/floatingMenu/elements/borderWidthAndColor.js

```diff
@@ -63,25 +63,23 @@ function BorderWidthAndColor() {
   const updateSelectedElements = useStory(
     (state) => state.actions.updateSelectedElements
   );
-
-  // Only multi-border elements support border opacity
-  const canHaveBorderOpacity = canSupportMultiBorder({ mask });
-
   // We only allow editing the current border width, if all borders are identical
   const hasUniformBorder =
     border.left === border.right &&
     border.left === border.top &&
     border.left === border.bottom;
 
-  // We only allow editing border color, if at least one border has a non-zero width
-  const hasBorderWidth = getHasBorderWidth(border);
+  // Border width and color inputs should only be rendered if all sides of the border are the same.
+  // Both checks are needed since not all shaped have lockedWidth eg. circles.
+  if (!border.lockedWidth && !hasUniformBorder) {
+    return null;
+  }
 
-  // If both controls are displayed, also add a dash between them
-  const hasBoth = hasUniformBorder && hasBorderWidth;
+  // Only multi-border elements support border opacity
+  const canHaveBorderOpacity = canSupportMultiBorder({ mask });
 
-  // NB: Note that it can never be the case, that both of the above bools, hasUniformBorder
-  // and hasBorderWidth, are false. If so, neither input would be shown. But because they
-  // partially contradict each other, one of the inputs will always render.
+  // We only allow editing border color, if at least one border has a non-zero width
+  const hasBorderWidth = getHasBorderWidth(border);
```

Comment:
💭 would be great to add some unit tests here to make sure the input renders when it's supposed to

Reply:
Yeah I wasn't sure if I should add some since the requirements might change?

Reply:
I can add them though.

Reply:
Sure! Let's add something simple for now. If/when the requirements change then we can come back and remove those.

Reply:
There were some tests already. Fixed ones needed the new logic and added one more.